src/rules/no_irregular_whitespace.rs

```diff
@@ -0,0 +1,62 @@
+use super::{Context, LintRule};
+use regex::Regex;
+use std::borrow::Cow;
+use std::sync::Arc;
+use swc_common::BytePos;
+use swc_ecmascript::ast::Module;
+
+pub struct NoIrregularWhitespace;
+
+fn test_for_whitespace(value: &Cow<str>) -> bool {
+  lazy_static! {
+    static ref ALL_IRREGULARS: Regex = Regex::new(r"[\f\v\u0085\ufeff\u00a0\u1680\u180e\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200a\u200b\u202f\u205f\u3000\u2028\u2029]").unwrap();
+    static ref LINE_BREAK_MATCHER: Regex = Regex::new(r"[^\r\n]+").unwrap();
+  }
+  ALL_IRREGULARS.is_match(value)
```

Comment:
This is good to test if there is some irregular whitespace 👍  after this check you can use [`Regex::find_iter`](https://docs.rs/regex/1.3.9/regex/struct.Regex.html#method.find_iter) to get `Matches` with all occurences and then use [`Match::range`](https://docs.rs/regex/1.3.9/regex/struct.Match.html#method.range) method to get indexes. 

After that you can create new span by adding start position of Regex's span. Check out [`Span::new`](https://docs.rs/swc_common/0.9.1/swc_common/struct.Span.html#method.new) method.

Reply:
Oh cool. I get it. Thanks